config/state_code_constraint.rb

```diff
@@ -3,12 +3,7 @@
 # License: AGPL-3.0-or-later WITH WTO-AP-3.0-or-later
 # Full license explanation at https://github.com/houdiniproject/houdini/blob/master/LICENSE
 class StateCodeConstraint
-  US_STATES = %w(AL AK AZ AR CA CO CT DE DC FL GA HI ID IL IN 
-      IA KS KY LA ME MD MA MI MN MS MO MT NE NV NH NJ 
-      NM NY NC ND OH OK OR PA PR RI SC SD TN TX UT VT VA WA WV WI WY)
-  US_STATES_REGEX = Regexp.union(*US_STATES.map{|i| /^#{i.downcase}$/})
-
   def matches?(request)
-    US_STATES_REGEX =~ request.params[:state_code]
+    ISO3166::Country[:US].subdivisions[request.params[:state_code].upcase] != nil
```

Comment:
I like this idea of using a gem here!

Assuming that subdivision is a `Hash` or a subclass of `Hash` (which I think it is), there's a better way to check if a key exists. `Hash` has a method called `has_key?` which accepts the key you want to search for as the first parameter.

That would change this to: `!ISO3166::Country[:US].subdivisions.has_key? request.params[:state_code].upcase`

Does this work?

Reply:
We swapped for it and it works! Thanks!